File: BMI/app/src/main/java/com/example/bmi/ShowCaloriesActivity.java
package com.example.bmi;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.TextView;

public class ShowCaloriesActivity extends AppCompatActivity {

    private TextView kcalResult, kcalError;
    private EditText getAge;
    private int radio, kg, cm;
    private int age = 0;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_show_calories);

        Button buttonCalculate = (Button) findViewById(R.id.buttonCalculateKcal);

        getAge = (EditText) findViewById(R.id.editTextAge);
        kcalResult = (TextView) findViewById(R.id.kcalResult);
        kcalError = (TextView) findViewById(R.id.kcalError);

        Intent intent = getIntent();
        kg = intent.getIntExtra("KG", 0);
        cm = intent.getIntExtra("CM", 0);

        final RadioGroup radioGroup = (RadioGroup) findViewById(R.id.kcalRadioGroup);
        radioGroup.setOnCheckedChangeListener(new RadioGroup.OnCheckedChangeListener()
        {
            @Override
            public void onCheckedChanged(RadioGroup group, int checkedId) {
                RadioButton rb = (RadioButton) findViewById(checkedId);
                // 1 - female, 2 - male
                radio = radioGroup.indexOfChild(rb);
            }
        });

        buttonCalculate.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                try{
                    kcalError.setText("");
                    age = Integer.parseInt(getAge.getText().toString());

                    if (radio == 1 && age != 0) {
                        int result = femaleKcalLimit();
                        kcalResult.setText(getString(R.string.kcal_result) + result);
                    }
                    else if (radio == 2 && age != 0) {
                        int result = maleKcalLimit();
                        kcalResult.setText(getString(R.string.kcal_result) + result);
                    }
                }catch (NumberFormatException err){
                    kcalError.setText(getString(R.string.text_error));
                }
            }
        });
    }

    protected int femaleKcalLimit () {
        return (int) (655.1 + (9.567 * kg) + (1.85 * cm) - (4.68 * age));
    }
    protected int maleKcalLimit () {
        return (int) (66.47 + (13.7 * kg) + (5.0 * cm) - (6.76 * age));
    }
}

File: BMI_kotlin/app/src/main/java/com/example/bmi_kotlin/ShowCaloriesActivity.kt
package com.example.bmi_kotlin

import android.os.Bundle
import android.util.Log
import android.widget.RadioButton
import android.widget.RadioGroup
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_show_calories.*

class ShowCaloriesActivity : AppCompatActivity() {

    var radio: String? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_show_calories)

        val bundle : Bundle = intent.extras!!
        val kg: Int = bundle.getInt("KG")
        val cm:Int = bundle.getInt("CM")
        
        kcalRadioGroup.setOnCheckedChangeListener(
                RadioGroup.OnCheckedChangeListener { group, checkedId ->
            val rb: RadioButton = findViewById(checkedId)
            radio = rb.text.toString()
        })

        buttonCalculateKcal.setOnClickListener {
            kcalError.text = ""
            kcalResult.text = ""
            if (radio.equals("Female") && editTextAge.text.toString().length > 0
                    && !editTextAge.text.toString().equals("0")) {
//                Log.v("AGE", age.toString())
                val fresult = femaleKcalLimit(kg, cm, editTextAge.text.toString().toInt())
                kcalResult.text = getString(R.string.kcal_result) + fresult
            }
            else if (radio.equals("Male") && editTextAge.text.toString().length > 0
                    && !editTextAge.text.toString().equals("0")) {
                val mresult = maleKcalLimit(kg, cm, editTextAge.text.toString().toInt())
                kcalResult.text = getString(R.string.kcal_result) + mresult
            }
            else kcalError.text = getString(R.string.text_error)
        }
    }

    fun femaleKcalLimit(kg: Int, cm: Int, age: Int): Int {
        return (655.1 + (9.567 * kg) + (1.85 * cm) - (4.68 * age)).toInt()
    }

    fun maleKcalLimit(kg: Int, cm: Int, age: Int): Int {
        return (66.47 + (13.7 * kg) + (5.0 * cm) - (6.76 * age)).toInt()
    }
}
